tracing-core/src/metadata.rs

```diff
@@ -208,12 +210,69 @@ impl<'a> fmt::Debug for Metadata<'a> {
         let mut meta = f.debug_struct("Metadata");
         meta.field("name", &self.name)
             .field("target", &self.target)
-            .field("level", &self.level);
+            .field("level", &self.level)
+            .field("location", &self.location)
+            .field("fields", &format_args!("{}", self.fields))
+            .field("callsite", &self.callsite())
+            .field("kind", &self.kind)
+            .finish()
+    }
+}
+
+// ===== impl Location =====
+impl<'a> Location<'a> {
+    /// Construct new source code location where the `Metadata` originated from with a file, line, column, module_path
+    pub const fn new(
+        file: Option<&'a str>,
+        line: Option<u32>,
+        column: Option<u32>,
+        module_path: Option<&'a str>,
+    ) -> Self {
+        Self {
+            file,
+            line,
+            column,
+            module_path,
+        }
+    }
+
+    /// Returns the column number in the source code file where this
+    /// `Metadata` originated from, or `None` if the column number is unknown.
+    pub fn column(&self) -> Option<u32> {
+        self.column
+    }
+
+    /// Returns the path to the Rust module where this
+    /// `Metadata` originated from, or `None` if the module path is unknown.
+    pub fn module_path(&self) -> Option<&'a str> {
+        self.module_path
+    }
+
+    /// Returns the name of the source code file where this
+    /// `Metadata` originated from, or `None` if the file is unknown
+    pub fn file(&self) -> Option<&'a str> {
+        self.file
+    }
+
+    /// Returns the line number in the source code file where this
+    /// `Metadata` originated from, or `None` if the line number is unknown.
+    pub fn line(&self) -> Option<u32> {
+        self.line
+    }
+}
+
+impl<'a> fmt::Debug for Location<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut meta = f.debug_struct("Location");
 
         if let Some(path) = self.module_path() {
             meta.field("module_path", &path);
         }
 
+        if let Some(column) = self.column() {
+            meta.field("column", &column);
+        }
+
         match (self.file(), self.line()) {
             (Some(file), Some(line)) => {
                 meta.field("location", &format_args!("{}:{}", file, line));
```

Comment:
it's a bit weird that this formats the column separately than the file and line number. i would expect it to either format them all as
```
<FILE>:<LINE>:<COLUMN>
```
or separately as fields, like
```
file: <FILE>,
line: <LINE>,
column: <COLUMN>,
```

Formatting the whole location together was initially done to make the `Metadata` struct's `Debug` output more concise. I would either add the column here, or change the `Debug` impl for `Location` to format them all as separate fields. If we did the second one, I'd then change it so that `Metadata` uses the single-line formatting for conciseness. We could add a `fmt::Display` impl for `Location` which does the single-line formatting, if you like --- some folks could probably find that useful?

Reply:
I made changes, let me know if it's good for you